dom/telephony/TelephonyDialCallback.cpp

```diff
@@ -104,6 +103,34 @@ TelephonyDialCallback::NotifyDialMMISuccessWithInteger(const nsAString& aStatusM
 }
 
 NS_IMETHODIMP
+TelephonyDialCallback::NotifyDialMMISuccessWithSession(const nsAString& aStatusMessage, uint32_t aServiceId)
+{
+  AutoJSAPI jsapi;
+  if (NS_WARN_IF(!jsapi.Init(mWindow))) {
+    return NS_ERROR_FAILURE;
+  }
+
+  JSContext* cx = jsapi.cx();
+
+  RootedDictionary<MozMMIResult> result(cx);
+  result.mSuccess = true;
+  result.mServiceCode.Assign(mServiceCode);
+  result.mStatusMessage.Assign(aStatusMessage);
+
+  nsCOMPtr<nsITelephonyService> service = do_GetService(TELEPHONY_SERVICE_CONTRACTID);
+  nsRefPtr<USSDSession> session = new USSDSession(mWindow, service, aServiceId);
+
+  JS::Rooted<JS::Value> jsAdditionalInformation(cx);
+  if (!ToJSValue(cx, session, &jsAdditionalInformation)) {
+    JS_ClearPendingException(cx);
+    return NS_ERROR_TYPE_ERR;
+  }
+
+  result.mAdditionalInformation.Construct().SetAsObject() = &jsAdditionalInformation.toObject();
```

Comment:
Here we instantiate our USSDSession object, we convert it to Javascript, and assign it to the additionalInformation field of our MozMMIResult object (this was inspired from NotifyDialMMISuccessWithStrings and NotifyDialMMISuccessWithCallForwardingOptions)